app/models/user.rb

```diff
@@ -325,6 +325,9 @@ def self.find_or_create_external_user(attrs, auth_source_name)
         new_usergroups += auth_source.external_usergroups.includes(:usergroup).where(:name => external_groups).map(&:usergroup)
         user.update(Hash[attrs.select { |k, v| v.present? }])
         user.usergroups = new_usergroups.uniq
+      else
```

Comment:
This will disable external login if `authorize_login_delegation_auth_source_user_autocreate` setting is disabled. Perhaps before  the previous `if` we should add:
```ruby
      unless user.auth_source.is_a? AuthSourceExternal
        logger.info...
        return nil
      end
```